Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLParameterScan.py

```diff
@@ -0,0 +1,277 @@
+        self.progress.report('Loading default mask')
+        if load_default_mask:
+            LoadNexusProcessed(Filename=self.default_mask, OutputWorkspace=default_mask_input)
+
+        process_absorber, absorber_name = needs_processing(self.absorber, 'Absorber')
+        self.progress.report('Processing absorber')
+        if process_absorber:
+            SANSILLReduction(Run=self.absorber,
+                             ProcessAs='Absorber',
+                             NormaliseBy=self.normalise,
+                             OutputWorkspace=absorber_name)
+
+        process_container, container_name = needs_processing(self.container, 'Container')
+
+        self.progress.report('Processing container')
+        if process_container:
+            SANSILLReduction(Run=self.container,
+                             ProcessAs='Container',
+                             OutputWorkspace=container_name,
+                             AbsorberInputWorkspace=absorber_name,
+                             CacheSolidAngle=True,
+                             NormaliseBy=self.normalise)
+        # even if the samples are provided as arguments, the SANSILLReduction won't use them and rather take the already
+        # processed ws
+        SANSILLReduction(Run=self.sample,
+                         AbsorberInputWorkspace=absorber_name,
+                         ContainerInputWorkspace=container_name,
+                         SampleThickness=self.thickness,
+                         SensitivityInputWorkspace=sens_input,
+                         SensitivityOutputWorkspace=self.output_sens,
+                         MaskedInputWorkspace=mask_input,
+                         DefaultMaskedInputWorkspace=default_mask_input,
+                         NormaliseBy=self.normalise,
+                         OutputWorkspace=sort_x_axis_output)
+
+        instrument = mtd["__joined"].getInstrument()
+        detector = instrument.getComponentByName("detector")
+        if "detector-width" in detector.getParameterNames() and "detector-height" in detector.getParameterNames():
+            width = int(detector.getNumberParameter("detector-width")[0])
+            height = int(detector.getNumberParameter("detector-height")[0])
+        else:
+            width, height = 100, 100
+            logger.warning("Width or height not found for the instrument. {0}, {1} assumed.".format(width, height))
+
+        self.checkPixelY(height)
+        grouping = create_detector_grouping(self.pixel_y_min, self.pixel_y_max, width, height)
+        GroupDetectors(InputWorkspace=sort_x_axis_output,
+                       OutputWorkspace=self.output2D,
+                       GroupingPattern=grouping,
+                       Behaviour="Average")
+        Transpose(InputWorkspace=self.output2D, OutputWorkspace=self.output2D)
+
+        if instrument.getName() == 'D16':
+            ws = mtd[self.output2D]
+            convert_spectrum_to_theta(ws, detector)
```

Comment:
This is error prone, I think the best is not to transpose here, so group the detectors, convert the spectrum axis, then transpose at the end. If needed, we should rather add the InPlane2Theta option to ConvertSpectrumAxis algorithm.